optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+        int problemCount = Runtime.getRuntime().availableProcessors() * 2;
+
+        SolverManager<TestdataSolution, Integer> solverManager = createSolverManagerTestableByDifferentConsumers();
+        assertDifferentSolveMethods(problemCount, solverManager);
+    }
+
+    private SolverManager<TestdataSolution, Integer> createSolverManagerTestableByDifferentConsumers() {
+        final SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(createGraduallyInitializedPhaseConfigs());
+
+        SolverManagerConfig solverManagerConfig = new SolverManagerConfig();
+
+        return SolverManager.create(solverConfig, solverManagerConfig);
+    }
+
+    private PhaseConfig[] createGraduallyInitializedPhaseConfigs() {
+        PhaseConfig[] configs = new PhaseConfig[2];
+        for (int x = 0; x < 2; x++) {
+            configs[x] = createSetXEntityToXValuePhaseConfig(x);
+        }
+        return configs;
+    }
+
+    private CustomPhaseConfig createSetXEntityToXValuePhaseConfig(int x) {
+        return new CustomPhaseConfig().withCustomPhaseCommands(
+                (ScoreDirector<TestdataSolution> scoreDirector) -> {
+                    TestdataSolution solution = scoreDirector.getWorkingSolution();
+                    TestdataEntity entity = solution.getEntityList().get(x);
+                    scoreDirector.beforeVariableChanged(entity, "value");
+                    entity.setValue(solution.getValueList().get(x));
+                    scoreDirector.afterVariableChanged(entity, "value");
+                    scoreDirector.triggerVariableListeners();
+                });
+    }
+
+    private void assertDifferentSolveMethods(int problemCount, SolverManager<TestdataSolution, Integer> solverManager) throws InterruptedException, ExecutionException {
+        assertSolveWithoutConsumer(problemCount, solverManager);
+        assertSolveWithConsumer(problemCount, solverManager, true);
+        assertSolveWithConsumer(problemCount, solverManager, false);
+    }
+
+    private void assertSolveWithoutConsumer(int problemCount, SolverManager<TestdataSolution, Integer> solverManager)
+            throws InterruptedException, ExecutionException {
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>(problemCount);
+
+        for (int id = 0; id < problemCount; id++) {
+            jobs.add(solverManager.solve(id, PlannerTestUtils.generateTestdataSolution(Integer.toString(id))));
+        }
+        assertInitializedJobs(jobs);
+    }
+
+    private void assertSolveWithConsumer(
+            int problemCount, SolverManager<TestdataSolution, Integer> solverManager, boolean onlyFinalBestSolutions)
+            throws ExecutionException, InterruptedException {
+
+        // Two solutions for every problem.
+        Map<Integer, List<TestdataSolution>> consumedSolutions = new HashMap<>(problemCount * 2);
```

Comment:
For consistency with other optaplanner code, the variable name of:
- Array and Collection uses plurals, for example "solutions"
- List uses a List suffix: for example  "solutionList"
- Map **definitly** uses a Map suffix, for example "solutionMap" (or "countToSolutionMap" in case the key is not dead obvious)